### PR TITLE
Dedicated function for ccm permissons

### PR DESCRIPTION
**Pull request by olemarkus**
The refactoring work from #11818

### CODE REVIEW

**Review by rifelpet — COMMENTED** [review]
A few comments, otherwise looks good to me though it does seem to increase the policy document size substantially so I worry users will hit size limits more frequently with `additionalPolicies`

File: pkg/model/iam/iam_builder.go [hunk 1]
```diff
@@ -759,6 +752,152 @@ func addCalicoSrcDstCheckPermissions(p *Policy) {
...
+		"ec2:CreateTags",
+	)
+}
+
+func AddCCMPermissions(p *Policy, clusterName string, cloudRoutes bool) {
+	resource := stringorslice.Slice([]string{"*"})
+
+	p.unconditionalAction.Insert(
+		"autoscaling:DescribeAutoScalingGroups",
+		"autoscaling:DescribeLaunchConfigurations",
```
Comment (johngmyers) [inline pkg/model/iam/iam_builder.go:803]: I wonder if we still need this.

Comment (olemarkus) [reply pkg/model/iam/iam_builder.go:803]: I don't think so. I couldn't see CCM use this. It is mentioned here, but this is probably dated https://cloud-provider-aws.sigs.k8s.io/prerequisites.html

**Comment by olemarkus** [comment]
> A few comments, otherwise looks good to me though it does seem to increase the policy document size substantially so I worry users will hit size limits more frequently with `additionalPolicies`

Looking at the strict json golden output, it will increase the size from roughly 5k to 6k. So substantially, but still a way to go before the limit is hit. The added bytes come from improving the cluster isolation, which I think is worth it.

File: pkg/model/iam/iam_builder.go [hunk 2]
```diff
@@ -759,6 +752,149 @@ func addCalicoSrcDstCheckPermissions(p *Policy) {
...
+	p.unconditionalAction.Insert(
+		"ec2:DescribeInstances", // aws.go
+	)
+}
+
+func addEtcdManagerPermissions(p *Policy) {
+	resource := stringorslice.Slice([]string{"*"})
+	p.unconditionalAction.Insert(
+		"ec2:DescribeVolumes", // aws.go
+
```
Comment (rifelpet) [inline pkg/model/iam/iam_builder.go:768]: nit:
```suggestion
```

otherwise LGTM

**Review by rifelpet — APPROVED** [review]

**Pull request merged**